### PR TITLE
Pulling in new api in sample app

### PR DESCRIPTION
**Pull request by ajaysubra**
⚠️ Merge only after #70 ⚠️
⚠️ Make sure to update the SDK from using branch to a version number once released ⚠️ 

This is still WIP, I went through and updated all the old API calls to the new one, but will review this on Monday. The idea is to know what the migration is, be able to write the migration script and merge this PR in when the updating to the new API happen.

### CODE REVIEW

File: Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift [hunk 1]
```diff
@@ -34,14 +34,14 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         window?.makeKeyAndVisible()
 
         // STEP2: Setup Klaviyo SDK with api key
-        Klaviyo.setupWithPublicAPIKey(apiKey: "magpcN")
+        KlaviyoSDK().initialize(with: "magpcN")
```
Comment (ndurell) [inline Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift:37]: Did you want to change this back to apiKey?

File: Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift [hunk 2]
```diff
@@ -34,14 +34,14 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
 
         // EXAMPLE: of how to track an event
-        Klaviyo.sharedInstance.trackEvent(eventName: "Opened kLM App")
+        KlaviyoSDK().create(event: .init(name: .CustomEvent("Opened kLM App")))
```
Comment (ndurell) [inline Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift:40]: You could keep an instance around if you want too.

File: Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift [hunk 3]
```diff
@@ -182,15 +170,19 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+// STEP6: Add this extension on AppDelegate for additional push notifications handling
 extension AppDelegate: UNUserNotificationCenterDelegate {
     // below method will be called when the user interacts with the push notification
     func userNotificationCenter(
         _ center: UNUserNotificationCenter,
         didReceive response: UNNotificationResponse,
         withCompletionHandler completionHandler: @escaping () -> Void) {
-        Klaviyo.sharedInstance.handlePush(userInfo: response.notification.request.content.userInfo as NSDictionary)
-        completionHandler()
+        // If this notifiation is Klaviyo's notification we'll handle it
```
Comment (ndurell) [inline Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift:180]: Nice!

**Review by ndurell — APPROVED** [review]
Looks good to me.

File: Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift (hunk 2 above)
Comment (ajaysubra) [reply Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift:40]: What is the best practice we want to suggest to our users? Keep an instance or use `KlaviyoSDK()` whenever they want to access the SDK.

File: Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift (hunk 1 above)
Comment (ajaysubra) [reply Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift:37]: I personally think it reads better as `KlaviyoSDK().initialize(with: "magpcN")` but it's not clear as to what initializing the SDK with.  Although, `KlaviyoSDK().initialize(apiKey: "magpcN")` doesn't read as well it's clear as to what the SDK requires to initialize.

**Pull request merged**